sources/packages/api/src/auth/guards/roles.guard.ts

```diff
@@ -21,6 +21,10 @@ export class RolesGuard implements CanActivate {
       return true;
     }
     const { user } = context.switchToHttp().getRequest();
-    return requiredRoles.some((role) => user.roles?.includes(role));
+    let userRoles;
```

Comment:
This can be changed as below:
```typescript

if (user.resource_access && user.azp) {
  const userRoles = user.resource_access[user.azp].roles;
  return requiredRoles.some((role) => userRoles?.includes(role));
}
return true;
```